src/Collection/Iterator/ExtractIterator.php

```diff
@@ -15,14 +15,19 @@
 namespace Cake\Collection\Iterator;
 
 use Cake\Collection\Collection;
+use Cake\Collection\CollectionInterface;
+use Cake\Collection\CollectionTrait;
+use IteratorIterator;
 
 /**
  * Creates an iterator from another iterator that extract the requested column
  * or property based on a path
  */
-class ExtractIterator extends Collection
+class ExtractIterator extends IteratorIterator implements CollectionInterface
 {
 
+    use CollectionTrait;
```

Comment:
It seems like the simpler option. There are lots of unserializable objects in PHP and CakePHP already.